File: app/src/main/java/com/bluetriangle/android/demo/java/screenTracking/ScreenLogFragment.java
package com.bluetriangle.android.demo.java.screenTracking;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.bluetriangle.analytics.screenTracking.ScreenTrackMonitor;
import com.bluetriangle.android.demo.DemoApplication;
import com.bluetriangle.android.demo.databinding.FragmentScreenLogsBinding;
import com.bluetriangle.android.demo.databinding.ScreenLogItemBinding;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class ScreenLogFragment extends Fragment {
    public static ScreenLogFragment newInstance() {
        return new ScreenLogFragment();
    }

    private FragmentScreenLogsBinding _binding;

    private final Adapter adapter = new Adapter();

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        _binding = FragmentScreenLogsBinding.inflate(inflater, container, false);
        return _binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        _binding = null;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        _binding.listView.setAdapter(adapter);

        adapter.setEntities(((DemoApplication) getActivity().getApplication()).screenTrackMonitor.getScreenLogs());
    }

    static class Adapter extends RecyclerView.Adapter<ViewHolder> {
        private final ArrayList<ScreenTrackMonitor.ScreenTrackLog> entities = new ArrayList<>();

        public void setEntities(List<ScreenTrackMonitor.ScreenTrackLog> entities) {
            this.entities.clear();
            this.entities.addAll(entities);
            notifyDataSetChanged();
        }

        @NonNull
        @Override
        public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            return new ViewHolder(ScreenLogItemBinding.inflate(LayoutInflater.from(parent.getContext())));
        }

        @Override
        public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
            holder.bind(entities.get(position));
        }

        @Override
        public int getItemCount() {
            return entities.size();
        }
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        ScreenLogItemBinding binding;

        public ViewHolder(ScreenLogItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(ScreenTrackMonitor.ScreenTrackLog data) {
            binding.txtScreenName.setText(data.getScreenName());
            binding.txtEventName.setText(data.isLoadEvent() ? "loaded on" : "viewed on");

            String startTime = new SimpleDateFormat("yyyy MMM dd, hh:mm:ss a", Locale.US).format(new Date(data.getStartTime()));
            float timeTaken = data.getTimeTaken() / 1000f;
            binding.txtEventTime.setText(String.format("%s for %.3f sec", startTime, timeTaken));
        }
    }
}

File: app/src/main/java/com/bluetriangle/android/demo/kotlin/screenTracking/ScreenLogFragment.kt
package com.bluetriangle.android.demo.kotlin.screenTracking

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import com.bluetriangle.analytics.screenTracking.ScreenTrackMonitor
import com.bluetriangle.android.demo.DemoApplication
import com.bluetriangle.android.demo.databinding.FragmentScreenLogsBinding
import com.bluetriangle.android.demo.databinding.ScreenLogItemBinding
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class ScreenLogFragment : Fragment() {
    companion object {
        fun newInstance() = ScreenLogFragment()
    }

    private var _binding: FragmentScreenLogsBinding? = null

    // This property is only valid between onCreateView and onDestroyView.
    private val binding get() = _binding!!
    private val adapter = Adapter()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentScreenLogsBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        _binding?.listView?.adapter = adapter

        adapter.setEntities((activity?.application as DemoApplication).scrTrack.screenLogs)
    }

    internal class Adapter : RecyclerView.Adapter<ViewHolder>() {
        private val entities: ArrayList<ScreenTrackMonitor.ScreenTrackLog> = arrayListOf()

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            return ViewHolder(
                ScreenLogItemBinding.inflate(LayoutInflater.from(parent.context))
            )
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            holder.bind(entities[position])
        }

        fun setEntities(entities: List<ScreenTrackMonitor.ScreenTrackLog>) {
            this.entities.clear()
            this.entities.addAll(entities)
            notifyDataSetChanged()
        }

        override fun getItemCount(): Int {
            return entities.size
        }
    }

    internal class ViewHolder(var binding: ScreenLogItemBinding) :
        RecyclerView.ViewHolder(binding.root) {
        @SuppressLint("SetTextI18n")
        fun bind(data: ScreenTrackMonitor.ScreenTrackLog) {
            binding.txtScreenName.text = data.screenName
            binding.txtEventName.text = if (data.isLoadEvent) "loaded on" else "viewed on"

            val startTime =
                SimpleDateFormat("yyyy MMM dd, hh:mm:ss a", Locale.US).format(Date(data.startTime))
            val timeTaken = data.timeTaken / 1000f
            binding.txtEventTime.text = "$startTime for $timeTaken sec"
        }
    }
}